### PR TITLE
Point generated column

### PR DESCRIPTION
**Pull request by cureseven**
#1

```
Jul 18 10:21:35 ip-172-31-39-90 isuumo[13082]: {"time":"2022-07-18T10:21:35.370878008Z","id":"","remote_ip":"127.0.0.1","host":"localhost:1323","method":"POST","uri":"/api/estate/nazotte","user_agent":"isucon-verify","status":500,"error":"","latency":37452962,"latency_human":"37.452962ms","bytes_in":386,"bytes_out":0}
Jul 18 10:21:35 ip-172-31-39-90 isuumo[13082]: {"time":"2022-07-18T10:21:35.380800483Z","level":"ERROR","prefix":"echo","file":"main.go","line":"890","message":"db access is failed on executing validate if estate is in polygon : Error 3037: Invalid GIS data provided to function st_geometryfromtext."}
```

### CODE REVIEW

File: webapp/go/main.go [hunk 1]
```diff
@@ -881,8 +881,7 @@ func searchEstateNazotte(c echo.Context) error {
 	for _, estate := range estatesInBoundingBox {
 		validatedEstate := Estate{}
 
-		point := fmt.Sprintf("'POINT(%f %f)'", estate.Latitude, estate.Longitude)
-		query := fmt.Sprintf(`SELECT * FROM estate WHERE id = ? AND ST_Contains(ST_PolygonFromText(%s), ST_GeomFromText(%s))`, coordinates.coordinatesToText(), point)
+		query := fmt.Sprintf(`SELECT id, name, description, thumbnail, address, latitude, longitude, rent, door_height, door_width, features, popularity FROM estate WHERE id = ? AND ST_Contains(ST_PolygonFromText(%s), ST_GeomFromText(estate_point))`, coordinates.coordinatesToText())
```
Comment (pinkumohikan) [inline webapp/go/main.go]: > ST_GeomFromText(estate_point)

estate_pointは既にPoint型なので `ST_GeomFromText()` に噛ます必要がない

**Comment by cureseven** [comment]
```
2022/07/18 10:48:14 bench.go:102: 最終的な負荷レベル: 3
{"pass":true,"score":872,"messages":[],"reason":"OK","language":"go"}
```

**Comment by cureseven** [comment]
```
mysql> explain SELECT id, name, description, thumbnail, address, latitude, longitude, rent, door_height, door_width, features, popularity FROM estate WHERE id = 11061 AND ST_Contains(ST_PolygonFromText('POLYGON((34.393587 135.193334,34.624634 134.987462,34.804558 134.874846,35.115623 135.184475,34.812190 135.245291,34.594194 135.233785,34.393587 135.193334))'), estate_point)\G;
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: estate
   partitions: NULL
         type: const
possible_keys: PRIMARY,idx_estate_point
          key: PRIMARY
      key_len: 4
          ref: const
         rows: 1
     filtered: 100.00
        Extra: NULL
1 row in set, 1 warning (0.00 sec)

ERROR:
No query specified
```

**Comment by cureseven** [comment]
```
Top 20 Sort By Total
Count    Total    Mean  Stddev    Min  P50.0  P90.0  P95.0  P99.0    Max   2xx  3xx  4xx  5xx  TotalBytes   MinBytes  MeanBytes   MaxBytes  Request
 1984  232.186  0.1170  0.1135  0.013  0.078  0.264  0.345  0.496  1.426  1974    0   10    0    30528565          0      15387      17284  GET /api/estate/search?$queryParams
 1193  220.482  0.1848  0.1286  0.020  0.135  0.378  0.444  0.620  0.992  1183    0   10    0    15450018          0      12950      28527  GET /api/chair/search?$queryParams
  394   78.575  0.1994  0.3453  0.000  0.051  0.518  0.892  1.957  2.001   389    0    5    0     6405200          0      16256      34133  POST /api/estate/nazotte HTTP/1.1
  313   38.701  0.1236  0.0857  0.006  0.098  0.251  0.309  0.406  0.509   311    0    2    0     4165186          0      13307      13742  GET /api/recommended_estate/<num> HTTP/1.1
 3674   19.018  0.0052  0.0082  0.000  0.004  0.010  0.014  0.032  0.235  3671    0    3    0     2253280          0        613        722  GET /api/estate/<num> HTTP/1.1
 2831   16.121  0.0057  0.0052  0.000  0.004  0.011  0.015  0.025  0.058  2827    0    4    0     1535204          0        542        636  GET /api/chair/<num> HTTP/1.1
    3    4.736  1.5787  1.0009  0.213  1.939  2.584  2.584  2.584  2.584     3    0    0    0           0          0          0          0  POST /api/estate HTTP/1.1
  835    3.420  0.0041  0.0094  0.001  0.003  0.006  0.010  0.022  0.236   835    0    0    0    11136765      13322      13337      13340  GET /api/estate/low_priced HTTP/1.1
  835    3.416  0.0041  0.0087  0.001  0.003  0.007  0.011  0.027  0.218   835    0    0    0    10129022      12121      12130      12156  GET /api/chair/low_priced HTTP/1.1
    3    3.415  1.1383  0.7002  0.197  1.343  1.875  1.875  1.875  1.875     3    0    0    0           0          0          0          0  POST /api/chair HTTP/1.1
    1    2.892  2.8920  0.0000  2.892  2.892  2.892  2.892  2.892  2.892     1    0    0    0          23         23         23         23  POST /initialize HTTP/1.1
  707    1.285  0.0018  0.0031  0.000  0.001  0.004  0.006  0.011  0.052   707    0    0    0           0          0          0          0  POST /api/estate/req_doc/<num> HTTP/1.1
  154    0.895  0.0058  0.0069  0.001  0.002  0.016  0.022  0.034  0.037   154    0    0    0           0          0          0          0  POST /api/chair/buy/<num> HTTP/1.1
  289    0.232  0.0008  0.0010  0.000  0.001  0.002  0.002  0.005  0.010   289    0    0    0      740707       2563       2563       2563  GET /api/estate/search/condition HTTP/1.1
  158    0.138  0.0009  0.0009  0.000  0.001  0.002  0.002  0.004  0.006   158    0    0    0      535936       3392       3392       3392  GET /api/chair/search/condition HTTP/1.1
```

**Comment by cureseven** [comment]
```
[CallStack]
    [scenario.verifyEstateNazotte] /home/isucon/isuumo/bench/scenario/verifyWithSnapshot.go:442
    [scenario.verifyWithSnapshot.func10] /home/isucon/isuumo/bench/scenario/verifyWithSnapshot.go:638
    [runtime.goexit] /home/isucon/local/go/src/runtime/asm_amd64.s:1373
2022/07/18 11:00:47 fails.go:105: [scenario.verifyEstateNazotte] /home/isucon/isuumo/bench/scenario/verifyWithSnapshot.go:442
    message("POST /api/estate/nazotte: レスポンスが不正です")
    message("snapshot: ../initial-data/result/verification_data/estate_nazotte/54.json")
    code(error application)
```

**Comment by cureseven** [comment]
```
2022/07/18 11:10:45 fails.go:105: [scenario.verifyEstateNazotte] /home/isucon/isuumo/bench/scenario/verifyWithSnapshot.go:442
    message("POST /api/estate/nazotte: レスポンスが不正です")
    message("snapshot: ../initial-data/result/verification_data/estate_nazotte/71.json")
    code(error application)
```

**Comment by cureseven** [comment]
image

```
{
  "count": 7,
  "estates": [
    {
      "id": 28900,
      "thumbnail": "/images/estate/ec86789d9ffbb8f9c753670594d89a108e0d9c1b90b02f69e8a05e747af77a7f.png",
      "name": "ISUアパート 中島",
      "description": "階段教室のまえの、「演劇部第二回公演　村山槐多作　長田俊文演出　『ドモ又の死』」というポスターが映しだされる。",
      "latitude": 35.63551447255732,
      "longitude": 139.72884428088662,
      "address": "新潟県御蔵島村白金台32丁目4番7号 平河町コート013",
      "rent": 104437,
      "doorHeight": 93,
      "doorWidth": 40,
      "features": "宅配ボックス,角部屋,オール電化"
    },
    {
      "id": 27977,
      "thumbnail": "/images/estate/0fd49a4200b71475b934e54d9237d4dda6ec58a5880e6916635e9fc554d91b8f.png",
      "name": "山田 ISUビル",
      "description": "『何がなんだかさつぱりお母さんにはわけがわからない、何言つたんでもなんでもなし、一人であゝして怒つてるんだもの。",
      "latitude": 35.63142485418902,
      "longitude": 139.72779390570605,
      "address": "佐賀県富津市津久戸町1丁目23番15号",
      "rent": 183325,
      "doorHeight": 107,
      "doorWidth": 162,
      "features": "防音室"
    },
    {
      "id": 15577,
      "thumbnail": "/images/estate/22d5b9e10d74f74ca617b8c81a2160fa39faa40488dd33927ef26337a82f336d.png",
      "name": "鈴木 ISUマンション",
      "description": "生物は、わずか数種の爬虫類がいるだけで、まったく、水掻きをつけ藻をかぶって現われる、水棲人の棲所というに適わしいのである。",
      "latitude": 35.62811881740756,
      "longitude": 139.7234748887714,
      "address": "大阪府川崎市幸区東浅草11丁目10番17号 パレス中鉢石町855",
      "rent": 100328,
      "doorHeight": 176,
      "doorWidth": 126,
      "features": ""
    },
    {
      "id": 26217,
      "thumbnail": "/images/estate/64cd2fe6a37b87d5d353e5aba485dd276e172e896f4c3196eb37195e49a8d028.png",
      "name": "ISU遠藤レジデンス",
      "description": "と、カムポスの様子が急に変ったのに気がつくと、なんだか勝負の結果が危ぶまれるような気に、折竹もだんだんになってきた。",
      "latitude": 35.61826817102464,
      "longitude": 139.72870392906208,
      "address": "熊本県川崎市宮前区四番町14丁目6番14号 渡辺パレス719",
      "rent": 181723,
      "doorHeight": 39,
      "doorWidth": 98,
      "features": "フローリング"
    },
    {
      "id": 21519,
      "thumbnail": "/images/estate/20cbaa801f6e506462ae1459f17009af4ec22f9ec223c72843777ccab11a226d.png",
      "name": "ISUアパート 橋本",
      "description": "バイエルタールの言葉を聞いていると、ときどき他のことを急にいいだすような、意想奔逸とみられるところが少なくない。",
      "latitude": 35.61555387553676,
      "longitude": 139.70866111961513,
      "address": "広島県印旛郡印旛村山中新田29丁目9番7号 シティ千塚652",
      "rent": 52223,
      "doorHeight": 185,
      "doorWidth": 185,
      "features": "ペット相談可,インターネット無料"
    },
    {
      "id": 10020,
      "thumbnail": "/images/estate/4a6403ece41cdf867fb3105e2e6b8139e4e41542874d1608d0b92af7e749f7ab.png",
      "name": "藤井 ISUマンション",
      "description": "男爵は、昨日仙臺に行く途中ふとこの町に泊らうといふ氣になつて降りたのだといつて、どてら姿で薄い髮の毛に櫛の齒を入れてゐるところであつた。",
      "latitude": 35.61545384486186,
      "longitude": 139.7067858799402,
      "address": "佐賀県羽村市白金台26丁目2番13号",
      "rent": 165421,
      "doorHeight": 43,
      "doorWidth": 91,
      "features": "女性限定,専用庭,ケーブルテレビ"
    },
    {
      "id": 1506,
      "thumbnail": "/images/estate/20c3343dfaee064fe36e29ec331947dabf153bc89ad90dcf3457c88f72c34b0a.png",
      "name": "ISU鈴木レジデンス",
      "description": "死体の全身はコチコチに硬直していて、すでに死後十時間は十分経過しているものと思われたが、さすが法水は動ぜずに、あくまで科学的批判を忘れなかった。",
      "latitude": 35.610867975321895,
      "longitude": 139.70851733913818,
      "address": "大分県調布市浅草橋22丁目2番15号",
      "rent": 51025,
      "doorHeight": 181,
      "doorWidth": 191,
      "features": ""
    }
  ]
}

```

**Comment by cureseven** [comment]
```
{
  "count": 5,
  "estates": [
    {
      "id": 15577,
      "thumbnail": "/images/estate/22d5b9e10d74f74ca617b8c81a2160fa39faa40488dd33927ef26337a82f336d.png",
      "name": "鈴木 ISUマンション",
      "description": "生物は、わずか数種の爬虫類がいるだけで、まったく、水掻きをつけ藻をかぶって現われる、水棲人の棲所というに適わしいのである。",
      "latitude": 35.62811881740756,
      "longitude": 139.7234748887714,
      "address": "大阪府川崎市幸区東浅草11丁目10番17号 パレス中鉢石町855",
      "rent": 100328,
      "doorHeight": 176,
      "doorWidth": 126,
      "features": ""
    },
    {
      "id": 1506,
      "thumbnail": "/images/estate/20c3343dfaee064fe36e29ec331947dabf153bc89ad90dcf3457c88f72c34b0a.png",
      "name": "ISU鈴木レジデンス",
      "description": "死体の全身はコチコチに硬直していて、すでに死後十時間は十分経過しているものと思われたが、さすが法水は動ぜずに、あくまで科学的批判を忘れなかった。",
      "latitude": 35.610867975321895,
      "longitude": 139.70851733913818,
      "address": "大分県調布市浅草橋22丁目2番15号",
      "rent": 51025,
      "doorHeight": 181,
      "doorWidth": 191,
      "features": ""
    },
    {
      "id": 10020,
      "thumbnail": "/images/estate/4a6403ece41cdf867fb3105e2e6b8139e4e41542874d1608d0b92af7e749f7ab.png",
      "name": "藤井 ISUマンション",
      "description": "男爵は、昨日仙臺に行く途中ふとこの町に泊らうといふ氣になつて降りたのだといつて、どてら姿で薄い髮の毛に櫛の齒を入れてゐるところであつた。",
      "latitude": 35.61545384486186,
      "longitude": 139.7067858799402,
      "address": "佐賀県羽村市白金台26丁目2番13号",
      "rent": 165421,
      "doorHeight": 43,
      "doorWidth": 91,
      "features": "女性限定,専用庭,ケーブルテレビ"
    },
    {
      "id": 21519,
      "thumbnail": "/images/estate/20cbaa801f6e506462ae1459f17009af4ec22f9ec223c72843777ccab11a226d.png",
      "name": "ISUアパート 橋本",
      "description": "バイエルタールの言葉を聞いていると、ときどき他のことを急にいいだすような、意想奔逸とみられるところが少なくない。",
      "latitude": 35.61555387553676,
      "longitude": 139.70866111961513,
      "address": "広島県印旛郡印旛村山中新田29丁目9番7号 シティ千塚652",
      "rent": 52223,
      "doorHeight": 185,
      "doorWidth": 185,
      "features": "ペット相談可,インターネット無料"
    },
    {
      "id": 26217,
      "thumbnail": "/images/estate/64cd2fe6a37b87d5d353e5aba485dd276e172e896f4c3196eb37195e49a8d028.png",
      "name": "ISU遠藤レジデンス",
      "description": "と、カムポスの様子が急に変ったのに気がつくと、なんだか勝負の結果が危ぶまれるような気に、折竹もだんだんになってきた。",
      "latitude": 35.61826817102464,
      "longitude": 139.72870392906208,
      "address": "熊本県川崎市宮前区四番町14丁目6番14号 渡辺パレス719",
      "rent": 181723,
      "doorHeight": 39,
      "doorWidth": 98,
      "features": "フローリング"
    }
  ]
}

```

**Comment by cureseven** [comment]
```
2022/07/18 11:23:17 bench.go:102: 最終的な負荷レベル: 5
{"pass":false,"score":1097,"messages":[{"text":"POST /api/chair: リクエストに失敗しました (critical error)","count":1}],"reason":"致命的なエラーが発生しました","language":"go"}
```

**Pull request merged**